app.rb

```diff
@@ -37,10 +37,18 @@ def search
 route :get, :post, '/api/create' do
   url = params[:url]
   password = params[:password]
-  begin
-    key = Haikunator.haikunate(0)
-    link = Link.get(key)
-  end until link.nil?
+  custom_key = params[:alias]
+
+  if custom_key
+    key = custom_key
+    halt 500, 'Duplicate Key' if Link.get(key)
```

Comment:
`4xx` seems to be more appropriate here. Let's think on what's the best status code to return in this use-case. What about `409`?

Reply:
Or `422`?